### PR TITLE
Adding Fallback to Yahoo! API

### PR DESCRIPTION
**Pull request by gabrielboliveira**
If the first API fail, it will try to get the exchange rate from Yahoo.

### CODE REVIEW

**Comment by gabrielboliveira** [comment]
@wesleyhf @dbpolito

File: bill.js [hunk 1]
```diff
@@ -17,6 +17,46 @@ const slackHook = restify.createJsonClient({url: env.slackWebhookURL});
...
+            if(dolarErr) {
+                tryFallback =  true;
+            } else if(obj && obj[0] && obj[0].bid) {
+                resolve(obj[0].bid);
+            } else {
+                tryFallback =  true;
+            }
+            if(tryFallback) {
+                getExchangeRateFallback().then(resolve, reject);
+            }
```
Comment (wesleyhf) [inline bill.js]:
````js
if (obj && obj[0] && obj[0].bid) {
    resolve(obj[0].bid);
} else {
    getExchangeRateFallback().then(resolve, reject);
}
````

File: bill.js [hunk 2]
```diff
@@ -59,15 +99,17 @@ schedule.scheduleJob(env.cron.schedule, () => {
...
-        slackHook.post('', {text}, (err, req, res, obj) => {
-            console.log(`Message sent. Status code -> ${res.statusCode}`);
-        });
-    });
+    getExchangeRate().then(
+        (exchangeRate) => {
+            let text = messageReplace(msg, {
+                'from': env.currency.from,
+                'to': env.currency.to,
+                'value': obj[0].bid
```
Comment (wesleyhf) [inline bill.js]: use `exchangeRate` var here

File: bill.js [hunk 3]
```diff
@@ -59,15 +99,17 @@ schedule.scheduleJob(env.cron.schedule, () => {
...
+            });
+            slackHook.post('', {text}, (err, req, res, obj) => {
```
Comment (wesleyhf) [inline bill.js]: enter

File: bill.js [hunk 4]
```diff
@@ -17,6 +17,46 @@ const slackHook = restify.createJsonClient({url: env.slackWebhookURL});
 const dolarAPI = restify.createJsonClient({
     url: `https://economia.awesomeapi.com.br/json/${env.currency.from}-${env.currency.to}/1`
 });
+const fallbackDolarAPI = restify.createJsonClient({
+    url: `http://download.finance.yahoo.com/d/quotes.csv?e=.csv&f=sl1d1t1&s=${env.currency.from}${env.currency.to}=X`
```
Comment (wesleyhf) [inline bill.js]: test the error message changing the url

Comment (gabrielboliveira) [inline bill.js]: Tested, working fine

image

**Pull request merged**